homeassistant/components/airpatrol/climate.py

```diff
@@ -0,0 +1,321 @@
+"""Climate platform for AirPatrol integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    SWING_OFF,
+    SWING_ON,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import AirPatrolDataUpdateCoordinator
+
+PARALLEL_UPDATES = 0
+
+_LOGGER = logging.getLogger(__name__)
+
+# HVAC modes supported by AirPatrol
+HVAC_MODES = {
+    "heat": HVACMode.HEAT,
+    "cool": HVACMode.COOL,
+    "off": HVACMode.OFF,
+}
+
+# Fan speeds supported by AirPatrol
+FAN_MODES = {
+    "min": FAN_LOW,
+    "max": FAN_HIGH,
+    "auto": FAN_AUTO,
+}
+
+# Swing modes supported by AirPatrol
+SWING_MODES = {
+    "on": SWING_ON,
+    "off": SWING_OFF,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up AirPatrol climate entities."""
+    coordinator: AirPatrolDataUpdateCoordinator = config_entry.runtime_data
+    # Create climate entities for each unit
+    entities = []
+    client = await coordinator.async_get_api_client()
+    units = await client.get_data()
+    entities = [
+        AirPatrolClimate(coordinator, unit, unit["unit_id"])
+        for unit in units
+        if "climate" in unit
+    ]
+
+    async_add_entities(entities)
+
+
+class AirPatrolClimate(
+    CoordinatorEntity[AirPatrolDataUpdateCoordinator], ClimateEntity
```

Comment:
Since I know you probably are going to add other platforms, perhaps add a base entity already in this PR which is subclassing the coordinator entity.

Reply:
I think I managed to get this done correctly!